instaclustr/resource_cluster.go

```diff
@@ -325,29 +365,165 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 		createData.RackAllocation = &rackAllocation
 	}
 
-	var jsonStr []byte
-	jsonStr, err = json.Marshal(createData)
+	var jsonStrCreate []byte
+	jsonStrCreate, err = json.Marshal(createData)
 	if err != nil {
 		return formatCreateErrMsg(err)
 	}
 
-	id, err := client.CreateCluster(jsonStr)
+	id, err := client.CreateCluster(jsonStrCreate)
 	if err != nil {
 		return formatCreateErrMsg(err)
 	}
 	d.SetId(id)
 	d.Set("cluster_id", id)
 	log.Printf("[INFO] Cluster %s has been created.", id)
-	return nil
+
+	kafkaSchemaRegistryUserPassword := d.Get("kafka_schema_registry_user_password").(string)
+	kafkaRestProxyUserPassword := d.Get("kafka_rest_proxy_user_password").(string)
+	waitForClusterState := d.Get("minimum_required_cluster_state").(string)
+
+	isKafkaCluster := false
+	hasRestProxy := false
+	hasSchemaRegistry := false
+
+	for i := 0; i < len(bundles); i++ {
+
+		if bundles[i].Bundle == "KAFKA" {
+			isKafkaCluster = true
+		}
+		if bundles[i].Bundle == "KAFKA_REST_PROXY" {
+			hasRestProxy = true
+		}
+		if bundles[i].Bundle == "KAFKA_SCHEMA_REGISTRY" {
+			hasSchemaRegistry = true
+		}
+	}
+
+	if len(waitForClusterState) > 0 {
```

Comment:
There should be some validation included to enforce waiting for the cluster to be running when the passwords are provided.